### ISSUE TITLE
Bug: RuleTester not correctly works with Typed Rules

### ISSUE THREAD

**Issue by pumano** [typescript-eslint/typescript-eslint#11784]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

rule-tester

### Repro Code

When I try to write unit tests for custom rule with types, I got error while trying to just instantiate `new RuleTester();` 

```bash
Error while loading rule '@rule-tester/my-custom-rule-with-types': You have used a rule which requires type information, but don't have parserOptions set to generate type information for this file. See https://typescript-eslint.io/getting-started/typed-linting for enabling linting with type information.
    Parser: (unknown)
    Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward "parserOptions.project" to @typescript-eslint/parser.
    Occurred while linting file.ts
```

but when I try to use `languageOptions` in `RuleTester` I got type error, because that `RuleTester` types don't contain any info about it:

example docs for typed rule testing:
https://typescript-eslint.io/developers/custom-rules/#testing-typed-rules
docs for types:
https://typescript-eslint.io/packages/rule-tester/#ruletester-constructor-options

my spec file looks like:

```TypeScript
import { RuleTester } from '@typescript-eslint/rule-tester';

import { rule, RULE_NAME } from './my-custom-rule-with-types';

const ruleTester = new RuleTester({
  languageOptions: {
    parserOptions: {
      projectService: {
        allowDefaultProject: ['*.ts*'],
      },
      tsconfigRootDir: __dirname,
    },
  },
});

ruleTester.run(RULE_NAME, rule, {
    valid: [],
    invalid: []
});
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

RuleTester should works with typed rules correctly

### Actual Result

RuleTester don't work as expected with typed rules

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/rule-tester`       | `8.47.0` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `8.47.0` |
| `TypeScript`                           | `5.7.3` |
| `ESLint`                               | `9.39.1` |
| `node`                                 | `22.21.0` |

**Comment by JoshuaKGoldberg** [comment]
> my spec file looks like:

We'll need more information than a snippet of one file. Please post a reproduction we can quickly clone/open and see the failure on. Without that, there's really nothing we can do to help - RuleTester _does_ work correctly with typed rules, as evidenced by this repo and many community plugins successfully using it. Thanks!

https://antfu.me/posts/why-reproductions-are-required

**Comment by pumano** [comment]
currently I solve problem via `import {RuleTester} from '@angular-eslint/test-utils';` that type of RuleTester solve problem. Currently I don't have some time to fully reproduce it via dedicated example.
